src/components/Login.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import axios from 'axios'
+
+import "./Login.css";
+import FormElement from './FormElement';
+import { useAuth } from './AuthContext';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLock, faUser } from '@fortawesome/free-solid-svg-icons'
+
+
+export const Login = () => {
+  const navigate = useNavigate();
+
+  // Redirect logged in user to dashboard
+  useEffect( () => {
+    let isLoggedIn = sessionStorage.getItem("isLoggedIn");
+    if (isLoggedIn){
+      navigate("/dashboard");
+    }
+  });
+
+  // Initialise variable state to empty string
+  const [formData, setFormData] = useState({
+    user_name: '',
+    user_password:'',
+  });
+
+  // Import global variables to update the login status
+  const {setLoginStatusGlobally} = useAuth();
+
+  const [error, setError] = useState('');
+
+  const handleChange = (event) => {
+    setFormData({ ...formData, 
+      [event.target.name]: event.target.value });
+  };
+
+  const handleSubmit = async (event) => {
+
+    event.preventDefault();
+
+    /*TODO:Implement comprehensive form validation using any framework!*/
+    if (!formData.user_name){
+      setError('Provide a valid username!');
+      return;
+    }
+    if (!formData.user_password){
+      setError('Provide a valid password!');
+      return;
+    }
+
+    try{
+      console.log('Data to be sent', formData); //debugging
+      // POST Login data
+      const response = await axios.post('/user/login', formData)
+      console.log(response); // debugging
+      // Set error to empty string upon success
+      setError('');
+      // Update global variable on sucessful login
+      setLoginStatusGlobally(true);
+      navigate("/dashboard")
+    }catch (error) {
+      /*
+       * Assuming the backend server sends the error message in the following 
+       * format: { "status":"error", "data": {}, "error":{"code": 400, "message
+       * ": "error message:}}
+       */
+      console.log('error', error); // debugging
+      if (error.response.data.error.message){ // debugging
```

Comment:
`response.data` gives a null pointer error, perhaps `response?.data` ? but I see that you use something like this on other lines as well